backend/src/core/services/stats.service.ts

```diff
@@ -1,19 +1,76 @@
-import { fn, cast, Transaction, Op } from 'sequelize'
+import { fn, cast, Transaction, Op, QueryTypes } from 'sequelize'
 import { Statistic, JobQueue, Campaign } from '@core/models'
 import {
   CampaignStats,
   CampaignStatsCount,
   CampaignInvalidRecipient,
 } from '@core/interfaces'
-import { MessageStatus } from '@core/constants'
+import { MessageStatus, ChannelType } from '@core/constants'
+import logger from '@core/logger'
+
+/**
+ * Calls update_stats SQL function for respective channels.
+ * fails silently
+ * @param campaignId
+ */
+const updateStats = async (campaignId: number): Promise<void> => {
+  const campaign = await Campaign.findOne({
+    where: { id: campaignId },
+  })
+  if (!campaign) return
+
+  logger.info(`updateStats invoked for campaign ${campaignId}`)
+
+  switch (campaign.type) {
```

Comment:
Could we place these in their respective *channel*-stats.service.ts please

Reply:
The reason why it's here is because if we ever have a consolidation script, you want to call all channels anyway, and you're just moving the switch case elsewhere?

I'm thinking from the POV of a cron job, if we need to find all campaigns where we need to run this updateStats function, will still need a `switch ... case` on the channel type to know which function to call right?

Reply:
The entire purpose of structuring our code base into core, email, sms, telegram is that core doesnt need to know anything about types. That being said, we should also extract getTwilioCredential and getTelegramCredential from backend/src/core/services/credential.service.ts